proto/iotics/api/circle.proto

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2019-2024 Iotic Labs Ltd. All rights reserved.
+
+// Iotics Web protocol definitions (circles)
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+import "iotics/api/common.proto";
+package iotics.api;
+
+option csharp_namespace = "Iotics.Api";
+option go_package = "github.com/Iotic-Labs/iotic-go-proto-qapi/iotics/api;ioticsapi";
+option java_multiple_files = true;
+option java_outer_classname = "CircleProto";
+option java_package = "com.iotics.api";
+option objc_class_prefix = "IAX";
+option php_namespace = "Iotics\\Api";
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+
+// CircleAPI enables creation and management of Iotics circles.
+// Services only affect local resources.
+service CircleAPI {
+
+  // UpsertCircle creates or update a circle with its metadata.
+  // The full state is applied (ie. if the operation succeeds the state of the circle will
+  // be the one described in the payload)
+  rpc UpsertCircle(UpsertCircleRequest) returns (UpsertCircleResponse) {}
+
+  // DeleteCircle deletes a circle.
+  rpc DeleteCircle(DeleteCircleRequest) returns (DeleteCircleResponse) {}
+
+  // Describes a circle.
+  rpc DescribeCircle(DescribeCircleRequest) returns (DescribeCircleResponse) {}
+
+  // List all circles.
+  rpc ListAllCircles(ListAllCirclesRequest) returns (ListAllCirclesResponse) {}
+}
+
+message CircleID {
+  // Circle Identifier (using DID format)
+  string id = 1;
+}
+
+
+message UpsertCircleRequest {
```

Comment:
(unimportant) some of the other definitions also have a docstring for each message type. (So I added them for e.g. the ACL PR). But I also appreciate that they are self-explanatory.

Reply:
Yes same. 
We prefer to go with consistency over "better" approach for the logic but we can stop adding redundant comments